src/libclipper/src/query_processor.cpp

```diff
@@ -212,46 +219,50 @@ boost::future<FeedbackAck> QueryProcessor::update(FeedbackQuery feedback) {
     predict_tasks = tasks_and_state.first.first;
     feedback_tasks = tasks_and_state.first.second;
     serialized_state = tasks_and_state.second;
-    log_info(
-        LOGGING_TAG_QUERY_PROCESSOR, "Used NewestModelSelectionPolicy to select tasks during feedback");
+    log_info(LOGGING_TAG_QUERY_PROCESSOR,
+             "Used NewestModelSelectionPolicy to select tasks during feedback");
   } else if (feedback.selection_policy_ == "simple_policy") {
     auto tasks_and_state = select_feedback_tasks<SimplePolicy>(
         feedback, query_id, get_state_table());
     // TODO: clean this up
     predict_tasks = tasks_and_state.first.first;
     feedback_tasks = tasks_and_state.first.second;
     serialized_state = tasks_and_state.second;
-    log_info(LOGGING_TAG_QUERY_PROCESSOR, "Used SimplePolicy to select tasks during feedback");
+    log_info(LOGGING_TAG_QUERY_PROCESSOR,
+             "Used SimplePolicy to select tasks during feedback");
   } else if (feedback.selection_policy_ == "bandit_policy") {
     auto tasks_and_state = select_feedback_tasks<BanditPolicy>(
         feedback, query_id, get_state_table());
     // TODO: clean this up
     predict_tasks = tasks_and_state.first.first;
     feedback_tasks = tasks_and_state.first.second;
     serialized_state = tasks_and_state.second;
-    log_info(LOGGING_TAG_QUERY_PROCESSOR, "Used BanditPolicy to select tasks during feedback");
+    log_info(LOGGING_TAG_QUERY_PROCESSOR,
+             "Used BanditPolicy to select tasks during feedback");
   } else {
-    log_error_formatted(
-        LOGGING_TAG_QUERY_PROCESSOR, "{} is an invalid feedback selection policy", feedback.selection_policy_);
+    log_error_formatted(LOGGING_TAG_QUERY_PROCESSOR,
+                        "{} is an invalid feedback selection policy",
+                        feedback.selection_policy_);
     // TODO better error handling
     return boost::make_ready_future(false);
   }
   log_info_formatted(LOGGING_TAG_QUERY_PROCESSOR,
-      "Scheduling {} prediction tasks and {} feedback tasks",
-      predict_tasks.size(), feedback_tasks.size());
+                     "Scheduling {} prediction tasks and {} feedback tasks",
+                     predict_tasks.size(), feedback_tasks.size());
 
   // 1) Wait for all prediction_tasks to complete
   // 2) Update selection policy
   // 3) Complete select_policy_update_promise
   // 4) Wait for all feedback_tasks to complete (feedback_processed future)
 
   // copy the vector
-  vector<shared_future<Output>> predict_task_completion_futures =
+  vector<boost::shared_future<Output>> predict_task_completion_futures =
       task_executor_.schedule_predictions({predict_tasks});
 
   vector<boost::future<FeedbackAck>> feedback_task_completion_futures =
       task_executor_.schedule_feedback(std::move(feedback_tasks));
 
+  // TODO: replace with clipper::future implementation of when_all
```

Comment:
Ahh it was a TODO because I hadn't done it yet. Removed all uses of boost future continuations.